ols/src/cache/redis_cache.py

```diff
@@ -70,14 +81,33 @@ def insert_or_append(self, user_id: str, conversation_id: str, value: str) -> No
             value: The value to set.
 
         Raises:
-            OutOfMemoryError: If item is evicted when Redis allocated
-                memory is higher than maxmemory.
+            RedisConnectionError: If item is unable to update after REDIS_MAX_RETRY.
         """
         key = super().construct_key(user_id, conversation_id)
-
         old_value = self.get(user_id, conversation_id)
-        with self._lock:
-            if old_value:
-                self.redis_client.set(key, old_value + "\n" + value)
-            else:
-                self.redis_client.set(key, value)
+        retry_count = 0
+        while retry_count < constants.REDIS_MAX_RETRY:
+            logger.debug("Updating redis cache ")
+            with self._lock:
+                if old_value:
+                    old_value.append(value)
+                    try:
+                        self.redis_client.set(key, old_value)
+                        break
+                    except Exception as redis_err:
```

Comment:
not all errors make sense to retry, do you know what kinds of exceptions to expect?